WireUI/Sources/WireReusableUIComponents/AccountImageView/AvailabilityIndicatorView.swift

```diff
@@ -0,0 +1,157 @@
+        fatalError("init(coder:) is not supported")
+    }
+
+    // MARK: - Methods
+
+    private func setupSubviews() {
+        addSubview(shapeView)
+        setNeedsLayout()
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+
+        guard let availability else {
+            shapeView.backgroundColor = .none
+            layer.mask = nil
+            return
+        }
+
+        let diameter = min(bounds.width, bounds.height)
+        let baseCircleFrame = CGRect(
+            origin: .init(x: (bounds.width - diameter) / 2, y: (bounds.height - diameter) / 2),
+            size: .init(width: diameter, height: diameter)
+        )
+        shapeView.frame = baseCircleFrame
+        shapeView.layer.cornerRadius = diameter / 2
+
+        switch availability {
+
+        case .available:
+            shapeView.backgroundColor = availableColor
+            layer.mask = nil
+
+        case .away:
+            shapeView.backgroundColor = awayColor
+
+            // mask with another circle, so that a ring results
+            let center = CGPoint(x: bounds.width / 2, y: bounds.height / 2)
+            let radius = (baseCircleFrame.width / 2) * (1 - awayRelativeBorderSize)
+            let maskPath = UIBezierPath(rect: bounds)
+            maskPath.addArc(
+                withCenter: center,
+                radius: radius,
+                startAngle: 0,
+                endAngle: 2 * .pi,
+                clockwise: true
+            )
+            let maskLayer = CAShapeLayer()
+            maskLayer.path = maskPath.cgPath
+            maskLayer.fillRule = .evenOdd
+            layer.mask = maskLayer
+
+        case .busy:
+            shapeView.backgroundColor = busyColor
+
+            // mask with a rectangle
+            let maskPath = UIBezierPath(rect: bounds)
+            let rectangleWidth = baseCircleFrame.width * busyMaskRelativeRectangleWidth
+            let rectangleHeight = baseCircleFrame.height * busyMaskRelativeRectangleHeight
+            let rectangleFrame = CGRect(
+                x: (bounds.width - rectangleWidth) / 2,
+                y: (bounds.height - rectangleHeight) / 2,
+                width: rectangleWidth,
+                height: rectangleHeight
+            )
+            maskPath.append(UIBezierPath(rect: rectangleFrame))
+            let maskLayer = CAShapeLayer()
+            maskLayer.path = maskPath.cgPath
+            maskLayer.fillRule = .evenOdd
+            layer.mask = maskLayer
+        }
+    }
+}
+
+// MARK: - Previews
+
+struct AvailabilityIndicatorView_Previews: PreviewProvider {
+
+    static var previews: some View {
+        VStack {
+            AvailabilityIndicatorViewRepresentable(.none)
+            AvailabilityIndicatorViewRepresentable(.available)
+            AvailabilityIndicatorViewRepresentable(.away)
+            AvailabilityIndicatorViewRepresentable(.busy)
+        }
+        .background(Color(UIColor.systemGray2))
+    }
+}
```

Comment:
Same here regarding the Preview.

Reply:
I changed the code a bit to use the `#Preview` macro, but I wasn't able to make it more compact or get rid of the `UIViewRepresentable`.

Reply:
Looking good!